### ISSUE TITLE
add queue specific resource constraint/property for partitioned resources

### ISSUE THREAD

**Issue by garlick** [flux-framework/flux-core#4438]
Problem:  to implement simple "partitions", we can use the `jobspec-default` plugin to set queue-specific properties that match properties set in _R_ that designate a partition.   So for example, if the user requests a queue named "debug", and we could configure 
```toml
[queues.debug.policy.jobspec.defaults]
system.constraints.properties = [ "debug" ]
```
However,
- The jobspec `properties` constraint operator value is defined as an array, but `jobspec-default` would not set the array if it already exists in the jobspec.
- a user could set a queue specific property and then get access to a partition wtihout picking up the queue specific limits

See also [RFC 31](https://flux-framework.readthedocs.io/projects/flux-rfc/en/latest/spec_31.html)

**Comment by grondo** [comment]
We could define that queue constraints are combined with any user specified constraints via the `and` operator. Fluxion does not currently support `and`, but if we only use `properties` in the constraint, then if we append to any existing `constraints.properties` array, then it would effectively be the same.

That would solve both points above, because:

 * jobspec-default (or a different plugin) would be able to modify the user's constraint, so it could set new values in the array even if it already exists
 * If a user sets a queue specific property then submits a job to a _different_ queue, then the job will be marked as infeasible if the queue constraints were not overlapping.
 
Ideally, we'd be able to use the `and` operator so I may spend some time seeing how much work it would be to add support to Fluxion for `and` `or` and `not`

**Comment by garlick** [comment]
Closed by #4587
